### PR TITLE
Resolve error when registering for non-existant camp

### PR DESCRIPTION
**Pull request by twilkhoo**
## Notion ticket link

[Resolve error when registering for non-existant camp](https://www.notion.so/uwblueprintexecs/Navigate-to-404-page-when-registering-for-non-existant-camp-f6dc0daeff154b4a9f2e0054315d4f6f)

## Implementation description
* Added another case of a camp not found.

## Steps to test
1. Go to http://localhost:3000/register/camp/thiscampclearlydoesntexist (or any other camp id that doesn't exist) and you should see a message saying "Error: Camp not found. Please go back and try again."

## What should reviewers focus on?
* Functionality

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/CampCreation/index.tsx [hunk 1]
```diff
@@ -532,7 +532,13 @@ const CampCreationPage = (): React.ReactElement => {
         isScheduleSessionsFilled={isScheduleSessionsFilled}
         isRegistrationFormFilled={isRegistrationFormFilled}
       />
-      <Box w="100%" flex="1" mt={[0, "0 !important"]} pb={20}>
+      <Box
+        w="100%"
+        flex="1"
+        mt={[0, "0 !important"]}
+        pb={20}
+        overflowX="hidden"
```
Comment (twilkhoo) [inline frontend/src/components/pages/CampCreation/index.tsx:540]: Not related, but this disables the horizontal scroll that appears in camp creation.

File: frontend/src/components/pages/RegistrantExperience/index.tsx [hunk 2]
```diff
@@ -125,8 +130,10 @@ const RegistrantExperiencePage = (): React.ReactElement => {
   };
 
   useEffect(() => {
-    if (!isLoading.camp && !isLoading.waiver) setCurrentStep("selection");
-  }, [isLoading]);
+    if (!isLoading.camp && !isLoading.waiver) {
+      setCurrentStep(camp && waiver ? "selection" : "notfound");
```
Comment (twilkhoo) [inline frontend/src/components/pages/RegistrantExperience/index.tsx:134]: This is what we did in the previous version of this file (with the `camp && waiver`).

Comment (TW1234567) [reply frontend/src/components/pages/RegistrantExperience/index.tsx:134]: Is there current support for the notfound step? I see that we have the Notfound component but it doesn't seem to be used

**Review by TW1234567 — COMMENTED** [review]
Just have a small comment about the not found page. All else looks good!

File: frontend/src/components/pages/RegistrantExperience/index.tsx (hunk 2 above)
Comment (twilkhoo) [reply frontend/src/components/pages/RegistrantExperience/index.tsx]: Yeah there is, the default case will be invoked when we set `currentStep` to `notFound`. Without this, we would never reach the default case which resulted in the error. I've added a little comment to make this clear beside the default case in `getCurrentStep`.

File: frontend/src/components/pages/WaitlistExperience/WaitlistFooter.tsx [hunk 3]
```diff
@@ -63,6 +63,7 @@ const WaitlistFooter = ({
       <Button
         ref={nextBtnRef}
         width={{ sm: "95vw", md: "45vw", lg: "auto" }}
+        mr={{ sm: 0, lg: 4 }}
```
Comment (twilkhoo) [inline frontend/src/components/pages/WaitlistExperience/WaitlistFooter.tsx:66]: Also not related, just adds some margin (exists on the other, RegistrationFooter component).

**Comment by TW1234567** [comment]
Look Good To Me!

**Pull request merged**